sfge/src/main/java/com/salesforce/rules/ops/ProgressListenerImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.salesforce.rules.ops;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.salesforce.config.SfgeConfigProvider;
+import com.salesforce.graph.ApexPath;
+import com.salesforce.messaging.CliMessager;
+import com.salesforce.messaging.EventKey;
+import com.salesforce.rules.Violation;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Publishes realtime information to CLI on the progress of analysis.
+ */
+public class ProgressListenerImpl implements ProgressListener {
+
+    @VisibleForTesting static final String NONE_FOUND = "none found";
```

Comment:
As mentioned in scrum, having this many individual events feels like overkill for now, but the strategy itself is sound.